### DISCUSSION TITLE
Sidebar & Table of Contents aren't showing up properly on content placed in different folder than: /content/docs/

### DISCUSSION THREAD

**Discussion by ochrebasis** [thuliteio/doks#1305] [How-to guides] [answered]
# Issue:

The issue I'm encountering is that the table of contents and sidebar aren't appearing on content which is placed outside of the `/content/docs/` subfolder.

The about page content mentioned below when placed under `/content/docs/` behaves as expected.

Here's an example of what a misrendered webpage looks like:
![issue](https://github.com/user-attachments/assets/0fe0dc33-be32-4415-9208-721f2a523081) 

#### Has sidebar and toc:    
- /content/docs/page/_index.md
- /content/docs/page/example.md
#### Does not have sidebar and toc:    
- /content/about/about-us/_index.md
- /content/about/about-us/about-infosec-club.md
- /content/about/about-us/infosec-leadership-and-advisors.md

# Steps taken to fix issue:

1. I ran: `npm run create -- --kind docs about`
2. I edited my `params.toml` file as such:

```toml
# Nav
  sectionNav = ["docs", "about"] # ["docs"] (default) or list of sections (e.g. ["docs", "guides"])
```

According to [Doks documentation the Table of Contents should be automatically generated](https://getdoks.org/docs/basics/navigation/).
> The “On this page” (Table of Contents) section is automatically generated. You can set the levels you’d like to show in ./config/_default/markup.toml

I'm a bit stumped by this to be honest. I believe I've followed all the steps correctly to get this to work properly, at least according to what the Doks documentation recommends.

As to why I have "about" picked instead of "about-us" is because I want the current layout to have different subsections:

```
content
├── about
│   ├── about-us
│   │   ├── about-infosec-club.md
│   │   ├── _index.md
│   │   └── infosec-leadership-and-advisors.md
│   ├── _index.md
│   └── other
│       ├── _index.md
│       └── privacy-policy.md
```
I also tested running `npx shx rm -rf public resources .hugo_build.lock && npm run dev` as mentioned somewhere else on this github to no avail. Also, all content headers include `draft: false`.

I've seen a solution to this problem on another doks website which I'm using as a reference, but their solution involves creating new layouts and custom scss to go with it. This is something I'd like to avoid not only because I like Doks default styling but also that goes outside the scope of my current skillset.

Here's a link to my repository: https://github.com/bellevuecollegeinfosec/bcinfosec.club

**Comment by h-enk** [comment, marked as answer]
@ochrebasis, sorry for getting back this late (other priorities). Interesting website you have!

Just add `type: docs` to the frontmatter of your page.:


![2024-11-26_08-44-34](https://github.com/user-attachments/assets/8b649800-f9d2-411f-9f59-29885526831c)

For more background see the Hugo docs: [type](https://gohugo.io/content-management/front-matter/#type)

**Reply by h-enk** [reply]
I will add this to the Doks docs

**Reply by ochrebasis** [reply]
Cheers mate, thanks for the help! 😁
